File: backend/app/schemas/claim.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ExtractedClaimRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    source_document_id: uuid.UUID | None
    source_chunk_id: uuid.UUID | None
    exact_text: str
    claim_category: str
    express_or_implied: str
    audience: str | None
    evidence_status: str
    intended_use_alignment: str | None
    regulatory_status_alignment: str | None
    risk: str
    recommended_disposition: str
    proposed_replacement: str | None
    review_status: str
    created_at: datetime


class ExtractedClaimWithProject(ExtractedClaimRead):
    project_name: str


class ExtractedClaimUpdate(BaseModel):
    review_status: str

File: frontend/src/types/claim.ts
export interface ExtractedClaim {
  id: string;
  project_id: string;
  project_name: string;
  source_document_id: string | null;
  source_chunk_id: string | null;
  exact_text: string;
  claim_category: string;
  express_or_implied: string;
  audience: string | null;
  evidence_status: string;
  intended_use_alignment: string | null;
  regulatory_status_alignment: string | null;
  risk: string;
  recommended_disposition: string;
  proposed_replacement: string | null;
  review_status: string;
  created_at: string;
}
